dcmgr/spec/dcmgr/scheduler/host_node/piped_rules_spec.rb

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+
+require 'spec_helper'
+require_relative '../../endpoints/12.03/helper'
+
+describe Dcmgr::Scheduler::HostNode::PipedRules do
+  describe '#schedule' do
+    before do
+      Dcmgr::Configurations.dcmgr.parse_dsl do |me|
+        me.instance_eval('
+          service_type("std", "StdServiceType") {
+            host_node_scheduler(:PipedRules) do
+
+              through(:RequestParamToGroup) {
+                default "hng-shhost"
+
+                key "host_node_group"
+                pair "local", "hng-local"
+              }
+              through(:LeastUsageBy) {
+                key "memory_size"
+              }
+
+            end
+          }
+        ')
+      end
+
+      host_ids = host_groups.map do |hng|
+        hng.mapped_uuids.map do |h|
+          h.id
+        end
+      end
+
+      host_ids.each do |h|
+        ds = Dcmgr::Models::HostNode.where(id: h)
+        allow(Isono::Models::NodeState).to receive(:filter).with(state: 'online').and_return(ds)
+      end
```

Comment:
I see. But still, I think this will only set the host nodes in the last host node group to online.

I think maybe you should try to add one more test case that uses multiple host node groups and assigns instances to each group.